src/array_api_extra/_funcs.py

```diff
@@ -548,3 +561,347 @@ def sinc(x: Array, /, *, xp: ModuleType | None = None) -> Array:
         xp.asarray(xp.finfo(x.dtype).eps, dtype=x.dtype, device=x.device),
     )
     return xp.sin(y) / y
+
+
+_undef = object()
+
+
+class at:  # pylint: disable=invalid-name
+    """
+    Update operations for read-only arrays.
+
+    This implements ``jax.numpy.ndarray.at`` for all backends.
+
+    Parameters
+    ----------
+    x : array
+        Input array.
+    idx : index, optional
```

Comment:
you could link to https://data-apis.org/array-api/latest/API_specification/indexing.html in the notes and perhaps reference it here, to explain what an `index` is